models/meshsync_events.go

```diff
@@ -0,0 +1,209 @@
+package models
+
+import (
+	"github.com/layer5io/meshkit/broker"
+	"github.com/layer5io/meshkit/database"
+	"github.com/layer5io/meshkit/logger"
+	"github.com/layer5io/meshkit/utils"
+	meshsyncmodel "github.com/layer5io/meshsync/pkg/model"
+	"gorm.io/gorm"
+)
+
+const (
+	MeshsyncStoreUpdatesSubject = "meshery-server.meshsync.store"
+	MeshsyncRequestSubject      = "meshery.meshsync.request"
+)
+
+// TODO: Create proper error codes for the functionalities this struct implements
+
+type MeshsyncDataHandler struct {
+	broker    broker.Handler
+	dbHandler database.Handler
+	log       logger.Handler
+}
+
+func NewMeshsyncDataHandler(broker broker.Handler, dbHandler database.Handler, log logger.Handler) *MeshsyncDataHandler {
+	return &MeshsyncDataHandler{
+		broker:    broker,
+		dbHandler: dbHandler,
+		log:       log,
+	}
+}
+
+func (mh *MeshsyncDataHandler) Run() error {
+	storeSubscriptionStatusChan := make(chan bool)
+
+	go mh.subsribeToStoreUpdates(storeSubscriptionStatusChan)
+	go mh.subscribeToMeshsyncEvents()
+	err := mh.removeStaleObjects()
+	if err != nil {
```

Comment:
If there is an error here then we will listen on the `storeSubscriptionStatusChan` causing the entire `subsribeToStoreUpdates` routine to be blocked.